### PR TITLE
[vcpkg] Fix [boost] find_package for MSVC2013 / v120

### PR DESCRIPTION
**Pull request by ManuelKugelmann**
**Describe the pull request**
added conditional set(Boost_COMPILER "-vc120") to vcpkg.cmake

- What does your PR fix? 
Fixes issue #10669

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

### CODE REVIEW

File: scripts/buildsystems/vcpkg.cmake [hunk 1]
```diff
@@ -246,7 +246,11 @@ macro(${VCPKG_OVERRIDE_FIND_PACKAGE_NAME} name)
         unset(Boost_USE_STATIC_RUNTIME)
         set(Boost_NO_BOOST_CMAKE ON)
         unset(Boost_USE_STATIC_RUNTIME CACHE)
-        set(Boost_COMPILER "-vc140")
+        if("${CMAKE_VS_PLATFORM_TOOLSET}" STREQUAL "v120")
+			set(Boost_COMPILER "-vc120")
+		else()
+			set(Boost_COMPILER "-vc140")
+		endif()
```
Comment (dan-shaw) [inline scripts/buildsystems/vcpkg.cmake:253]: Can you fix the spacing?

**Review by dan-shaw — APPROVED** [review]

**Comment by ManuelKugelmann** [comment]
yes

**Comment by ManuelKugelmann** [comment]
@dan-shaw  @PhoebeHui Ready to merge?

**Comment by strega-nil** [comment]
What's going on with all these port failures?

**Pull request merged**